sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -387,6 +395,48 @@ public QueueClientBuilder clientOptions(ClientOptions clientOptions) {
         return this;
     }
 
+    /**
+     * Sets the queue message encoding.
+     *
+     * @param messageEncoding {@link QueueMessageEncoding}.
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code messageEncoding} is {@code null}.
+     */
+    public QueueClientBuilder messageEncoding(QueueMessageEncoding messageEncoding) {
+        this.messageEncoding = Objects.requireNonNull(messageEncoding, "'messageEncoding' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the handler that performs the tasks needed when a message is received or peaked from the queue
+     * but cannot be decoded.
+     * <p>
+     * Such message can be received or peaked when queue is expecting certain {@link QueueMessageEncoding}
+     * but there's another producer that is not encoding messages in expected way.
+     * I.e. the queue contains messages with different encoding.
+     * <p>
+     * {@link QueueMessageDecodingFailure} contains {@link QueueAsyncClient} for the queue that has received
+     * the message as well as {@link QueueMessageDecodingFailure#getQueueMessageItem()} or
+     * {@link QueueMessageDecodingFailure#getPeekedMessageItem()}  with raw body, i.e. no decoding will be attempted
+     * so that body can be inspected as has been received from the queue.
+     * <p>
+     * The handler won't attempt to remove the message from the queue. Therefore such handling should be included into
+     * handler itself.
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.queue.QueueClientBuilder#messageDecodingFailedHandler}
+     *
+     * @param messageDecodingFailedHandler the handler.
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code messageDecodingFailedHandler} is {@code null}.
+     */
+    public QueueClientBuilder messageDecodingFailedHandler(
+        Function<QueueMessageDecodingFailure, Mono<Void>> messageDecodingFailedHandler) {
+        this.messageDecodingFailedHandler = Objects.requireNonNull(messageDecodingFailedHandler,
```

Comment:
I noticed in the client code this was null checked and by default this is null in the builder, given that should null be an allowed value?